freqtrade/exchange/__init__.py

```diff
@@ -105,132 +94,231 @@ def validate_pairs(pairs: List[str]) -> None:
+        )
+    except ccxt.BaseError as e:
+        raise OperationalException(e)
 
 
 def get_balance(currency: str) -> float:
     if _CONF['dry_run']:
         return 999.9
 
-    return _API.fetch_balance()[currency]
+    # ccxt exception is already handled by get_balances
+    balances = get_balances()
+    return balances[currency]['free']
 
 
-def get_balances():
+def get_balances() -> dict:
     if _CONF['dry_run']:
-        return []
+        return {}
+
+    try:
+        balances = _API.fetch_balance()
+        # Remove additional info from ccxt results
+        balances.pop("info", None)
+        balances.pop("free", None)
+        balances.pop("total", None)
+        balances.pop("used", None)
+
+        return balances
+    except ccxt.NetworkError as e:
+        raise NetworkException(
+            'Could not get balance due to networking error. Message: {}'.format(e)
+        )
+    except ccxt.BaseError as e:
+        raise OperationalException(e)
 
-    return _API.fetch_balance()
 
-# @cached(TTLCache(maxsize=100, ttl=30))
 @retrier
 def get_ticker(pair: str, refresh: Optional[bool] = True) -> dict:
-    return _API.fetch_ticker(pair)
+    global _TICKER_CACHE
+    try:
+        if not refresh:
+            if _TICKER_CACHE and pair in _TICKER_CACHE:
+                return _TICKER_CACHE[pair]
+        _TICKER_CACHE[pair] = _API.fetch_ticker(pair)
+        return _TICKER_CACHE[pair]
+    except ccxt.NetworkError as e:
+        raise NetworkException(
+            'Could not load tickers due to networking error. Message: {}'.format(e)
+        )
+    except ccxt.BaseError as e:
+        raise OperationalException(e)
 
 
-# @cached(TTLCache(maxsize=100, ttl=30))
 @retrier
-def get_ticker_history(pair: str, tick_interval) -> List[List]:
-    # TODO: tickers need to be in format 1m,5m
-    # fetch_ohlcv returns an [[datetime,o,h,l,c,v]]
+def get_ticker_history(pair: str, tick_interval: str) -> List[Dict]:
     if 'fetchOHLCV' not in _API.has or not _API.has['fetchOHLCV']:
-        logger.warning('Exhange %s does not support fetching historical candlestick data.',
-                       _API.name)
-        return []
+        raise OperationalException(
+            'Exhange {} does not support fetching historical candlestick data.'.format(_API.name)
+        )
 
     try:
-        ohlcv = _API.fetch_ohlcv(pair, timeframe=str(tick_interval)+"m")
-        return ohlcv
+        history = _API.fetch_ohlcv(pair, timeframe=tick_interval)
+        history_json = []
+        for candlestick in history:
+            history_json.append({
+                'T': arrow.get(candlestick[0]/1000.0).strftime('%Y-%m-%dT%H:%M:%S.%f'),
+                'O': candlestick[1],
+                'H': candlestick[2],
+                'L': candlestick[3],
+                'C': candlestick[4],
+                'V': candlestick[5],
+            })
+        return history_json
```

Comment:
should we convert the format back to the bittrex format?
#576 - https://github.com/gcarq/freqtrade/pull/576#discussion_r176931604 points out to use the ccxt format as is as that seems to be the way to go.
@shusso comments?

Reply:
@xmatthias Yes. ccxt format is the way. No need to worry about the old one.

Reply:
What do you think about using the current format for now and moving to ccxt format in a later PR?

Reply:
i am not sure why - changing to the new format now doesn't really complicate things, feat/objectify-ccxt has most of it already handled, both dry-mode and backtesting seems to work (with refresh of the pair-data though).

I'm working on a "convert"-script which is almost ready (conversion is working fine - just not handling all cases yet), so existing test-data can be converted easily.

Reply:
The conversion script is good and ccxt format could be the new format. The advantage of the current format is that it is more human-readable. But I'd rather leave it for a different PR after ccxt is implemented, since it's a somewhat large change.

Reply:
I think it's easier to change the format same time as ccxt is taken into use. Does that happen in this PR or the next, I really don't mind. But like @xmatthias said, it's pretty much there already, so why change it back

Reply:
Moved to ccxt format. Next PR will update files.